File: dotnet/Vuln/Easy/OpenRedirectDirect.cs
namespace SastEval.Vuln.Easy;

/// <summary>
/// CWE-601: Open Redirect - User-supplied URL used directly in redirect without validation.
/// </summary>
public static class OpenRedirectDirect
{
    public static IResult Handle(HttpContext context)
    {
        var url = context.Request.Query["url"].ToString();

        if (string.IsNullOrEmpty(url))
        {
            return Results.BadRequest("Missing 'url' query parameter");
        }

        // VULNERABLE: Redirect to user-controlled URL without validation
        return Results.Redirect(url);
    }
}

File: java/src/main/java/com/sasteval/vuln/easy/OpenRedirectDirect.java
package com.sasteval.vuln.easy;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * CWE-601: Open Redirect (Direct)
 * VULNERABILITY: User-supplied URL is used in redirect without validation.
 */
public class OpenRedirectDirect extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        String url = request.getParameter("url");

        // VULN: Unvalidated redirect to user-supplied URL
        response.sendRedirect(url);
    }
}
